async-nats/src/jetstream/context.rs

```diff
@@ -90,6 +90,42 @@ impl Context {
             Response::Ok(info) => Ok(Stream { info }),
         }
     }
+
+    pub async fn delete_stream<T: AsRef<str>>(&self, stream: T) -> Result<bool, Error> {
+        let stream = stream.as_ref();
+        if stream.is_empty() {
+            return Err(Box::new(io::Error::new(
+                ErrorKind::InvalidInput,
+                "the stream name must not be empty",
+            )));
+        }
+        let subject = format!("{}.STREAM.DELETE.{}", self.prefix, stream);
+        let response: Response<DeleteStatus> = self.request(subject, &json!({})).await?;
+        match response {
```

Comment:
Nit, can just inline this?

```suggestion
        match self.request(subject, &json!({})).await? {
```